src/backend/distributed/executor/multi_utility.c

```diff
@@ -391,6 +399,25 @@ multi_ProcessUtility(Node *parsetree,
 	{
 		ListCell *ddlJobCell = NULL;
 
+		if (IsA(parsetree, AlterTableStmt))
```

Comment:
Should not we have this check in `ErrorIfUnsupportedAlterTableStmt`?

Looking this more, it looks like you do checks here after running master query so it becomes easier to use create table checks? Am I right?